include/circt/Dialect/SV/SVStatements.td

```diff
@@ -17,6 +17,17 @@ def isModuleSymbol : AttrConstraint<
       "&$_op, $_self.cast<::mlir::FlatSymbolRefAttr>().getValue()))"
   >, "is module like">;
 
+//===----------------------------------------------------------------------===//
+// Macro operations
+//===----------------------------------------------------------------------===//
+
+def DefineOp : SVOp<"define", [NonProceduralOp]> {
+  let summary = "'define MACRO' statement";
+
+  let arguments = (ins StrAttr:$macroName, OptionalAttr<StrAttr>:$value);
```

Comment:
Is this attribute just used for verbatim text? I'm curious if we'd need anything more interesting. I guess a string is a fine place to start.

Also, we always mention this, but an optional string can have three states: missing, present and empty, present and non-empty. I think the should just follow the convention we normally use, for consistency. I believe we have standardized on a non-optional string, and helper builders to create an op with an empty string if none is needed. I know this means we can't elide it from the assembly format declaratively, but I think we can use a `custom<>()` assembly format for this.